src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -111,7 +171,7 @@ static Record NodeByLabelScanConsume(OpBase *opBase) {
 	GrB_Index nodeId;
 	bool depleted = false;
 	GxB_MatrixTupleIter_next(op->iter, NULL, &nodeId, &depleted);
-	if(depleted) return NULL;
+	if(depleted || _outOfBounds(op, nodeId)) return NULL;
```

Comment:
I hope to get rid of this `_outOfBounds` check, by either not caring for the max id or by introducing a range tuple iterator.

Reply:
done